lib/src/Base/Stat/FractionalBrownianMotionModel.cxx

```diff
@@ -0,0 +1,241 @@
+  , exponent_(1, 0.25)
+  , eta_(0)
+  , rho_(0)
+{
+  // Nothing to do
+}
+
+/* Parameters constructor */
+FractionalBrownianMotionModel::FractionalBrownianMotionModel(const Scalar scale,
+    const Scalar amplitude,
+    const Scalar exponent)
+  : CovarianceModelImplementation(Point(1, scale), Point(1, amplitude))
+  , exponent_(0)
+{
+  setExponentEtaRho(Point(1, exponent), SquareMatrix(1), CorrelationMatrix(1));
+}
+
+
+/* Parameters constructor */
+FractionalBrownianMotionModel::FractionalBrownianMotionModel(const Scalar scale,
+    const Point & amplitude,
+    const Point & exponent,
+    const SquareMatrix & eta,
+    const CorrelationMatrix & rho)
+  : CovarianceModelImplementation(Point(1, scale), amplitude)
+{
+  setExponentEtaRho(exponent, eta, rho);
+}
+
+
+/* Virtual constructor */
+FractionalBrownianMotionModel * FractionalBrownianMotionModel::clone() const
+{
+  return new FractionalBrownianMotionModel(*this);
+}
+
+/* Computation of the covariance function
+ */
+CovarianceMatrix FractionalBrownianMotionModel::operator() (const Point & s,
+    const Point & t) const
+{
+  if (s.getDimension() != inputDimension_) throw InvalidArgumentException(HERE) << "Error: the point s has dimension=" << s.getDimension() << ", expected dimension=" << inputDimension_;
+  if (t.getDimension() != inputDimension_) throw InvalidArgumentException(HERE) << "Error: the point t has dimension=" << t.getDimension() << ", expected dimension=" << inputDimension_;
+  MatrixImplementation result(outputDimension_, outputDimension_);
+  const Scalar sOverTheta = s[0] / scale_[0];
+  const Scalar tOverTheta = t[0] / scale_[0];
+  const Scalar stOverTheta = tOverTheta - sOverTheta;
+  const Scalar absSOverTheta = std::abs(sOverTheta);
+  const Scalar absTOverTheta = std::abs(tOverTheta);
+  const Scalar absSTOverTheta = std::abs(stOverTheta);
+  const Scalar sgnSOverTheta = sOverTheta > 0.0 ? 1.0 : -1.0;
+  const Scalar sgnTOverTheta = tOverTheta > 0.0 ? 1.0 : -1.0;
+  const Scalar sgnSTOverTheta = stOverTheta > 0.0 ? 1.0 : -1.0;
+  std::cerr << "sOverTheta=" << sOverTheta << ", tOverTheta=" << tOverTheta << ", stOverTheta=" << stOverTheta << std::endl;
+  std::cerr << "absSOverTheta=" << absSOverTheta << ", absTOverTheta=" << absTOverTheta << ", absSTOverTheta=" << absSTOverTheta << std::endl;
+  for (UnsignedInteger i = 0; i < outputDimension_; ++i)
+  {
+    const Scalar Hi = exponent_[i];
+    const Scalar sigmaI = amplitude_[i];
+    std::cerr << "i=" << i << ", Hi=" << Hi << ", sigmaI=" << sigmaI << std::endl;
+    for (UnsignedInteger j = 0; j < i; ++j)
+    {
+      const Scalar Hij = Hi + exponent_[j];
+      const Scalar sPowHij = std::pow(absSOverTheta, Hij);
+      const Scalar tPowHij = std::pow(absTOverTheta, Hij);
+      const Scalar stPowHij = std::pow(absSTOverTheta, Hij);
+      const Scalar correlation = rho_(i, j) * (sPowHij + tPowHij - stPowHij);
+      Scalar asymmetry = 0.0;
+      if (std::abs(Hij - 1.0) < 1e-7)
```

Comment:
This constant is hardcoded, a comment on this ?

Reply:
I added a comment saying that this threshold gives full double precision (according to Maple)